### PR TITLE
Switch default package suffix to unstable for Linuxes

### PR DESCRIPTION
**Pull request by martyngigg**
Description of work.

Sets the default package suffix for linux builds to `unstable`.

**To test:**

Code review. Pay close attention to the buildscript changes.

Run cmake with `ENABLE_CPACK=ON` with a clean build to examine the effect of the changes. Check the package installation scripts have the ENVVARS_ON_INSTALL set as expected.

*No issue*

**Release Notes** 

*Does not need to be in the release notes.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: buildconfig/Jenkins/buildscript [hunk 1]
```diff
@@ -183,29 +183,22 @@ if [[ "$BUILDPKG" == true ]]; then
...
-    if [[ ${JOB_NAME} == *python3* ]]; then
-	PACKAGE_SUFFIX=${PACKAGE_SUFFIX}-python3
-    fi
-
-    # Only unsuffixed release builds create envvars scripts
-    if [[ ${JOB_NAME} == *release* && -z "$PACKAGE_SUFFIX" ]]; then
-      PACKAGINGVARS="${PACKAGINGVARS} -DENVVARS_ON_INSTALL=True -DCPACK_SET_DESTDIR=ON"
+    if [[ ${JOB_NAME} == *release* ]]; then
+      # No suffix and traditional install path
+      PACKAGINGVARS="${PACKAGINGVARS} -DCMAKE_INSTALL_PREFIX=/opt/Mantid -DCPACK_PACKAGE_SUFFIX="
```
Comment (rosswhitfield) [inline buildconfig/Jenkins/buildscript]: If we are having coexisting python 2 and 3 releases soon won't the packages need unique names?

**Review by rosswhitfield — CHANGES_REQUESTED** [review]

File: buildconfig/Jenkins/buildscript (hunk 1 above)
Comment (martyngigg) [reply buildconfig/Jenkins/buildscript]: Good point - in my head python 3 branch went through the else branch and became `mantid-python3` but of course that will have `release` in the name too. I'll fix that.

**Review by rosswhitfield — CHANGES_REQUESTED** [review]
You still have the case if the `JOB_NAME` has both `release` and `python3` you end up with `PACKAGE_SUFFIX=unstable-python3` when you just want `-python3`.

**Comment by martyngigg** [comment]
@rosswhitfield The python3/release behaviour should be fixed now.

**Review by rosswhitfield — APPROVED** [review]

**Pull request merged**